pintc/tests/optimizations/dead_state_elimination.pnt

```diff
@@ -136,4 +136,21 @@ predicate Test {
 //     constraint (::h == (::e_1 == ::e_0));
 //     constraint __eq_set(__mut_keys(), {{1, 0}, 2, {1, (0 + 1)}, 2, {0}, 1, 8});
 // }
+// >>>
+
+// warnings <<<
+// state is unused
+// @619..640: state is unused
+// state is unused
+// @646..669: state is unused
+// state is unused
+// @675..698: state is unused
+// state is unused
+// @704..729: state is unused
+// state is unused
+// @735..761: state is unused
+// state is unused
+// @767..793: state is unused
+// state is unused
+// @848..872: state is unused
```

Comment:
@mohammadfawaz You'll want to look at this. I have `DeadStates` being emitted as a warning. I'm not familiar enough with how `mut` works to know if we should be letting the user know that the `state` is unused or not. Currently it is being reported as a warning.

I left a `TODO:` above about this that we can discuss.

Reply:
Let's remove this warning for now and address it in a separate issue. We first need to make sure that there is no `mut` anywhere in the `state` initializer before we emit a warning.